tests/unit/dns_test.cc

```diff
@@ -150,3 +151,33 @@ static future<> test_srv() {
 SEASTAR_TEST_CASE(test_srv_tcp) {
     return test_srv();
 }
+
+
+SEASTAR_TEST_CASE(test_parallel_resolve_name) {
+    dns_resolver::options opts;
+    opts.use_tcp_query = true;
+
+    auto d = ::make_lw_shared<dns_resolver>(std::move(opts));
+    return when_all(
+        d->resolve_name("www.google.com"),
+        d->resolve_name("www.google.com"),
+        d->resolve_name("www.google.com"),
+        d->resolve_name("www.google.com"),
+        d->resolve_name("www.google.com"),
+        d->resolve_name("www.google.com")
+    ).finally([d](auto&&...) {}).discard_result();
+}
+
+SEASTAR_TEST_CASE(test_parallel_resolve_name_udp) {
+    dns_resolver::options opts;
+
+    auto d = ::make_lw_shared<dns_resolver>(std::move(opts));
+    return when_all(
+        d->resolve_name("www.google.com"),
+        d->resolve_name("www.google.com"),
+        d->resolve_name("www.google.com"),
+        d->resolve_name("www.google.com"),
+        d->resolve_name("www.google.com"),
+        d->resolve_name("www.google.com")
+    ).finally([d](auto&...) {}).discard_result();
+}
```

Comment:
Will the test fail before the fix? If so better reorder it (or squash into same commit), to avoid false positive during bisect.

Reply:
Done